### PR TITLE
Fix flaky golden test

### PR DESCRIPTION
**Pull request by chejennifer**
SV scores from sv detector can change a little between each run (after 6 decimal digits). So sort variables by truncating their score to 6 decimal digits and if variables have the same truncated score, sort alphabetically

### CODE REVIEW

File: server/integration_tests/explore_test.py [hunk 1]
```diff
@@ -235,463 +248,8 @@ def handle_i18n_response(self, resp, i18n_lang):
 
     self.assertTrue(success, f"wanted: {i18n_lang}, got {detected}")
 
-  def test_detection_basic(self):
```
Comment (shifucun) [inline server/integration_tests/explore_test.py:238]: revert these deletions?

File: server/integration_tests/explore_test.py [hunk 2]
```diff
@@ -235,463 +248,8 @@ def handle_i18n_response(self, resp, i18n_lang):
...
-                       check_detection=True)
-
-  def test_detection_context(self):
-    self.run_detection('detection_api_context', [
-        'States with highest PHDs', 'Commute in tracts of California',
-        'Compare with Nevada', 'Correlate with asthma',
-        'countries with greenhouse gas emissions',
-        'median income in Santa Clara county and Alameda county'
-    ])
-
```
Comment (pradh) [inline server/integration_tests/explore_test.py:307]: Don't delete all the tests? :)

File: server/integration_tests/explore_test.py [hunk 3]
```diff
@@ -111,6 +111,19 @@ def handle_response(self,
...
+    # runs. Sort by scores cut off after 6 digits after the decimal and for
+    # variables with the same truncated score, sort alphabetically
+    resp_var_to_score = {}
+    for i, sv in enumerate(dbg['sv_matching']['SV']):
+      score = dbg['sv_matching']['CosineScore'][i]
+      resp_var_to_score[sv] = float("{:.6f}".format(score))
+    if 'variables' in resp:
+      sorted_variables = sorted(resp['variables'],
+                                key=lambda x: (-resp_var_to_score[x], x))
+      resp['variables'] = sorted_variables
```
Comment (shifucun) [inline server/integration_tests/explore_test.py:125]: should all the golden scores change to 6 digit now?

File: server/integration_tests/explore_test.py (hunk 1 above)
Comment (chejennifer) [reply server/integration_tests/explore_test.py]: good catch, updated

File: server/integration_tests/explore_test.py (hunk 2 above)
Comment (chejennifer) [reply server/integration_tests/explore_test.py]: good catch, updated

File: server/integration_tests/explore_test.py (hunk 3 above)
Comment (chejennifer) [reply server/integration_tests/explore_test.py:125]: no, this just updates the ordering of variables

**Review by shifucun — APPROVED** [review]

**Review by pradh — APPROVED** [review]
Thanks!

Would the "proper fix" be to change NL Server to be more deterministic?  Should we have TODO?

**Comment by chejennifer** [comment]
> Thanks!
> 
> Would the "proper fix" be to change NL Server to be more deterministic? Should we have TODO?

added TODO

**Pull request merged**